opcodes/dxil/dxil_nvapi.cpp

```diff
@@ -258,6 +258,81 @@ static bool emit_nvapi_extn_op_get_special(Converter::Impl &impl)
 	return false;
 }
 
+static spv::Id bitcast_operand_to_float(Converter::Impl &impl, const LLVMBC::Value *op)
+{
+	auto *cast_inst = llvm::dyn_cast<llvm::CastInst>(op);
+
+	if (cast_inst && cast_inst->getOpcode() == llvm::Instruction::BitCast)
+	{
+		op = cast_inst->getOperand(0);
+
+		if (op->getType()->getTypeID() == llvm::Type::TypeID::FloatTyID)
+			return impl.get_id_for_value(op);
+	}
+
+	auto *bitcast_op = impl.allocate(spv::OpBitcast, impl.builder().makeFloatType(32));
+	bitcast_op->add_id(impl.get_id_for_value(op));
+	impl.add(bitcast_op);
+
+	return bitcast_op->id;
+}
+
+static bool emit_nvapi_extn_op_hit_object_make_miss(Converter::Impl &impl)
+{
+	spv::Id index;
+	if (!get_argument(impl, NVAPI_ARGUMENT_SRC0U + 0, &index))
+		return false;
+
+	auto *tmin = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC0U + 1];
+	auto *tmax = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC0U + 2];
+	auto *origin_x = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC1U + 0];
+	auto *origin_y = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC1U + 1];
+	auto *origin_z = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC1U + 2];
+	auto *direction_x = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC2U + 0];
+	auto *direction_y = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC2U + 1];
+	auto *direction_z = impl.nvapi.fake_doorbell_inputs[NVAPI_ARGUMENT_SRC2U + 2];
+
+	if (!tmin || !tmax || !origin_x || !origin_y || !origin_z || !direction_x || !direction_y || !direction_z)
+		return false;
+
+	auto &builder = impl.builder();
+
+	builder.addExtension("SPV_NV_shader_invocation_reorder");
+	builder.addCapability(spv::CapabilityShaderInvocationReorderNV);
+
+	spv::Id t_min = bitcast_operand_to_float(impl, tmin);
+	spv::Id t_max = bitcast_operand_to_float(impl, tmax);
+
+	spv::Id ray_origin[3] = {
```

Comment:
This is a little problematic for reproducible output. I think the arguments can be evaluated in arbitrary order. It'd be better to just

`for (int i = 0; i < 3; i++) ray_origin[i] = bitcast_operand_to_float(impl, impl.nvapi.fake_doorbell_inputs[SRC1U + i])`

etc.